backend/api/Areas/Tools/Helpers/ImportPropertiesHelper.cs

```diff
@@ -0,0 +1,410 @@
+                b_e.LocalId = property.LocalId;
+                b_e.Description = property.Description;
+                b_e.Latitude = double.Parse(property.Latitude);
+                b_e.Longitude = double.Parse(property.Longitude);
+                b_e.BuildingFloorCount = int.Parse(property.BuildingFloorCount);
+                b_e.BuildingTenancy = property.BuildingTenancy;
+
+                // Find foreign key.
+                var build_type = _buildingConstructionTypes.FirstOrDefault(bct => bct.Name == property.BuildingConstructionType);
+                var build_use = _buildingPredominateUses.FirstOrDefault(bpu => bpu.Name == property.BuildingPredominateUse);
+
+                // If the building construction type doesn't exist, create it.
+                if (build_type == null)
+                {
+                    var max_id = _buildingConstructionTypes.Max(pc => pc.Id) + 1;
+                    build_type = new Entity.BuildingConstructionType(max_id, property.BuildingConstructionType);
+                    _pimsAdminService.BuildingConstructionType.Add(build_type);
+                    _buildingConstructionTypes.Add(build_type);
+                }
+
+                // If the building predominate use doesn't exist, create it.
+                if (build_use == null)
+                {
+                    var max_id = _buildingPredominateUses.Max(pc => pc.Id) + 1;
+                    build_use = new Entity.BuildingPredominateUse(max_id, property.BuildingPredominateUse);
+                    _pimsAdminService.BuildingPredominateUse.Add(build_use);
+                    _buildingPredominateUses.Add(build_use);
+                }
+
+                b_e.BuildingConstructionTypeId = build_type.Id;
+                b_e.BuildingConstructionType = build_type;
+                b_e.BuildingPredominateUseId = build_use.Id;
+                b_e.BuildingPredominateUse = build_use;
+
+                var city = GetOrCreateCity(property.City);
+
+                // Add/Update the address.
+                if (b_e.AddressId == 0)
+                {
+                    _logger.LogDebug($"Adding address for building '{property.ParcelId}'-''{property.LocalId}'.");
+
+                    var address = new Entity.Address(property.CivicAddress, null, city.Id, "BC", property.Postal);
+                    b_e.Address = address;
+                }
+                else
+                {
+                    b_e.Address.Address1 = property.CivicAddress;
+                    b_e.Address.CityId = city.Id;
+                    b_e.Address.Postal = property.Postal;
+                }
+            }
+
+            // Add a new evaluation for each year.
+            if (!b_e.Evaluations.Any(e => e.FiscalYear == fiscal))
+            {
+                var evaluation = new Entity.BuildingEvaluation(fiscal, b_e)
+                {
+                AssessedValue = float.Parse(property.AssessedValue)
```

Comment:
I'm guessing this is the only field that is in the data?

Reply:
correct